pump/storage/sorter.go

```diff
@@ -149,6 +159,18 @@ func (s *sorter) pushTSItem(item sortItem) {
 func (s *sorter) run() {
 	defer s.wg.Done()
 
+	go func() {
+		// Avoid if no any more pushTSItem call so block at s.cond.Wait() in run() waiting the matching c-binlog
```

Comment:
When `sorter` get closed, `s.cond.Broadcast` is called to wake all goroutines waiting for `s.cond`, isn't that enough?

Reply:
we don't close it, it may block at s.cond.Wait case when no
more any item push.

Reply:
When closing the server, storage is closed and so is the sorter in it.

Reply:
not talking the closing the server case, if block at the no matching c-binlog case it will nerver wake up if not any more any item push (tidb may crash and no c-binlog write, we may stop to write fakebinlog when offline in waitUntilCommitTSSaved)

Reply:
I see.

But with the following goroutine added, we will be periodically waking up `Wait`s, which is just like when we were checking periodically. Then maybe we should just get rid of `s.cond` because it's no longer helping us?

Reply:
it can still make it wake up immediately once we push new item anc check for matching c-binog but not just check periodically